base_layer/core/src/proof_of_work/blake_pow.rs

```diff
@@ -45,11 +49,11 @@ impl BlakePow {
     /// A simple miner. It starts at nonce = 0 and iterates until it finds a header hash that meets the desired target
     // ToDo convert to future, with ability to break function. We need to be able to stop this if we receive a mined
     // block
-    pub fn mine(target_difficulty: Difficulty, header: &BlockHeader) -> u64 {
+    pub fn mine(target_difficulty: Difficulty, header: &BlockHeader, stop_flag: Arc<AtomicBool>) -> u64 {
         let mut nonce = 0u64;
         // We're mining over here!
         while let Ok(d) = header.pow.achieved_difficulty(nonce, &header) {
-            if d >= target_difficulty {
+            if d >= target_difficulty || (*stop_flag).load(Ordering::Relaxed) {
```

Comment:
Does this not compile?

```suggestion
            if d >= target_difficulty || stop_flag.load(Ordering::Relaxed) {
```